project_templates/fastapi_safir_app/example/src/example/config.py

```diff
@@ -2,39 +2,39 @@
 
 from __future__ import annotations
 
-import os
-from dataclasses import dataclass
+from pydantic import BaseSettings, Field
+from safir.logging import LogLevel, Profile
 
 __all__ = ["Configuration", "config"]
 
 
-@dataclass
-class Configuration:
+class Configuration(BaseSettings):
     """Configuration for example."""
 
-    name: str = os.getenv("SAFIR_NAME", "example")
-    """The application's name, which doubles as the root HTTP endpoint path.
-
-    Set with the ``SAFIR_NAME`` environment variable.
-    """
-
-    profile: str = os.getenv("SAFIR_PROFILE", "development")
-    """Application run profile: "development" or "production".
-
-    Set with the ``SAFIR_PROFILE`` environment variable.
-    """
-
-    logger_name: str = os.getenv("SAFIR_LOGGER", "example")
-    """The root name of the application's logger.
-
-    Set with the ``SAFIR_LOGGER`` environment variable.
-    """
-
-    log_level: str = os.getenv("SAFIR_LOG_LEVEL", "INFO")
-    """The log level of the application's logger.
-
-    Set with the ``SAFIR_LOG_LEVEL`` environment variable.
-    """
+    name: str = Field(
+        "example",
+        title="Name of application",
+        description="Doubles as the root HTTP endpoint path.",
+        env="SAFIR_NAME",
+    )
+
+    profile: Profile = Field(
+        Profile.development,
+        title="Application logging profile",
+        env="SAFIR_PROFILE",
+    )
+
+    logger_name: str = Field(
```

Comment:
I've lately been reconsidering the logger name thing. I think my original intent a long time ago was to truncate the logger down to just the app's name, but now that we've got JSON-formatted logs, we've got lots of tools to work with log output. So I'm wondering if the template should hard code the logger name to the package's name. Then `get_logger(__name__)` would always work.

In other words, I'm not sure there's a use-case for dynamically configuring the logger anymore, so maybe we should hard-code it in the source?

Reply:
Oh, sure, I can make that change. I have never set that configuration option anyway.

Reply:
It's convenient to have for setting the root logger, but yeah, it seems like something that never needs to be configured.